### DISCUSSION TITLE
What's the current compatible Ubuntu version for developing Dioxus desktop app?

### DISCUSSION THREAD

**Discussion by davidschmitt** [DioxusLabs/dioxus#1079] [Q&A]
First, thanks for creating Dioxus - it is just what I need for my first Rust application experiment!

After getting my initial desktop app to run under (containerized) Ubuntu with Dioxus 0.3.2, I ran into the event propagation problem reported here: https://github.com/DioxusLabs/dioxus/issues/1068

When I attempt to use the Dioxus master branch from GitHub as recommended in that discussion, now I see library version problems.  E.g.:

```
  cargo:warning=`PKG_CONFIG_ALLOW_SYSTEM_CFLAGS="1" "pkg-config" "--libs" "--cflags" "gio-2.0" "gio-2.0 >= 2.70"` did not exit successfully: exit status: 1
  error: could not find system library 'gio-2.0' required by the 'gio-sys' crate

  --- stderr
  Requested 'gio-2.0 >= 2.70' but version of GIO is 2.64.6
```

What version of Ubuntu (or other Linux distro) should I use to reduce the risk of such problems in the (near) future?

**Comment by DogeDark** [comment]
Looks like you might just need to update 'gio'
I'd try the following command: ``sudo apt-get install libglib2.0-dev`` and see if Dioxus compiles

**Reply by davidschmitt** [reply]
I ran that, but I'm already on the latest version of 'gio' for Ubuntu 20.04 apparently:

```
$ sudo apt-get install libglib2.0-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libglib2.0-dev is already the newest version (2.64.6-1~ubuntu20.04.4).
libglib2.0-dev set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

Since I'm running containerized Ubuntu I'm just trying to match the version being used by Dioxus developers.   Since 20.04 LTS doesn't work I expect it's one of these:

- Ubuntu 23.04
- Ubuntu 22.10
- Ubuntu 22.04.2 LTS
- Ubuntu 22.04.1 LTS
- Ubuntu 22.04 LTS

So what's my best choice?

**Reply by DogeDark** [reply]
I'm not entirely sure. I'd assume the latest LTS should have 'gio' >= 2.70

**Reply by davidschmitt** [reply]
Just FYI, I tried Ubuntu 22.04 and got it to work.  So I'm good to go for now, but the Dioxus docs should probably mention the recommended/supported Linux distros/releases.
